### PR TITLE
URIs: remove repeated id.

### PR DESCRIPTION
**Pull request by htuch**
It's a _lot_ more convenient when writing proto literals (e.g. in
filesystem config files) to be able to specify a single flat ID than
repeated. Also, this makes some of the decisions around /-escaping
simpler in different implementations.

Signed-off-by: Harvey Tuch <htuch@google.com>

### CODE REVIEW

File: udpa/core/v1/resource_locator.proto [hunk 1]
```diff
@@ -40,10 +40,10 @@ message ResourceLocator {
   // URI scheme.
   Scheme scheme = 1 [(validate.rules).enum = {defined_only: true}];
 
-  // Opaque identifiers for the resource. These are effectively concatenated
-  // with ‘/’ to form the non-query param path as resource ID. This may end
+  // Opaque identifier for the resource. Any '/' will not be escaped during URI
+  // encoding and will form part of the URI path. This may end
   // with ‘*’ for glob collection references.
-  repeated string id = 2 [(validate.rules).repeated = {min_items: 1}];
+  string id = 2 [(validate.rules).string = {min_len: 1}];
```
Comment (markdroth) [inline udpa/core/v1/resource_locator.proto:46]: Is `min_len: 1` correct?  Could this be empty?  Or this string expected to have a leading `/`?

**Review by markdroth — COMMENTED** [review]
Looks great!  Just one minor comment about the expected value.

File: udpa/core/v1/resource_locator.proto (hunk 1 above)
Comment (htuch) [reply udpa/core/v1/resource_locator.proto:46]: Yeah, I think you are right, we do allow empty resource IDs, will fix.

Comment (markdroth) [reply udpa/core/v1/resource_locator.proto]: Should we explicitly say that the value is not expected to contain a leading `/`?

Comment (htuch) [reply udpa/core/v1/resource_locator.proto]: I don't think so; the URL encoding scheme in the comment above already states this.

File: udpa/core/v1/resource_name.proto [hunk 2]
```diff
@@ -15,17 +15,17 @@ option (udpa.annotations.file_status).work_in_progress = true;
 
 // UDPA resource name. This has a canonical udpa:// URI representation:
 //
-//   udpa://{authority}/{type_url}/{id/*}?{context_params}
+//   udpa://{authority}/{type_url}/{id}?{context_params}
```
Comment (markdroth) [inline udpa/core/v1/resource_name.proto:18]: Need to also remove the `/*` in two places in resource_locator.proto.

**Review by markdroth — APPROVED** [review]
Looks great, modulo a couple of remaining comment updates.

**Review by markdroth — APPROVED** [review]

**Pull request merged**